### ISSUE TITLE
AS events of a single sample

### ISSUE THREAD

**Issue by biochristmas** [Xinglab/rmats-turbo#418]
Hi, If I only need to obtain the AS count for a single sample (experimental group or control group), I execute this command: rmats.py --b1 b2.txt --gtf isoquant_gffcompare.gtf -t paired --bi ./STAR_binary_index --readLength 150 --nthread 30 --od ./test1 --libType fr-unstranded --statoff --tmp ./test1/tmp_output. The AS count in the resulting summary.txt file represents the number of alternative splicing events for that sample？
![summary](https://github.com/user-attachments/assets/21d4a3c1-71ec-4ce1-b93d-ab3a68c263ca)

**Comment by EricKutschera** [comment]
The `TotalEvents` columns in the summary file are the total number of events in the MATS output files: https://github.com/Xinglab/rmats-turbo/issues/257#issuecomment-1419471666

Since you only have `--b1` without `--b2` the events are not filtered and could be from the `--gtf` without supporting reads: https://groups.google.com/g/rmats-user-group/c/SVlOJs4CwVk/m/IIzvDAXrBgAJ

You could get an AS event count by filtering on the read count (IJC_SAMPLE_1, SJC_SAMPLE_1) and PSI (IncLevel1) columns: https://github.com/Xinglab/rmats-turbo/issues/183#issuecomment-1054318031

**Comment by biochristmas** [comment]
Hi, @EricKutschera I will filter based on Average PSI (IncLevel) within 0.05 and 0.95. I have three replicates, as an example (first row of the image), Average PSI (IncLevel) = (NA+1.0+0.0)/3. However, there are NA values present. How should I handle this。
![SE MATS JC](https://github.com/user-attachments/assets/625dfa46-80a0-4983-944f-b8f5a9db774a)

**Comment by EricKutschera** [comment]
You can try different methods for handling NA. One option is to filter out events that have any NA value in IncLevel1. It looks like only a few rows in the screenshot have NA IncLevel1. You could also keep events with only 1 NA value in IncLevel1 and average the remaining values. In the example you mentioned you could use `(1 + 0)/2=0.5` as the average PSI

**Comment by biochristmas** [comment]
Hi, @EricKutschera
I have another question. IncLevel1 = SJC_SAMPLE_1 / (SJC_SAMPLE_1 + IJC_SAMPLE_2). For example, in the second row of the image, the value for the third replicate should be 0.06: 2/(2+31)=0.06. However, IncLevel1 = 0.037.

**Comment by biochristmas** [comment]
Hi, @EricKutschera 
I have another question. IncLevel1 = SJC_SAMPLE_1 / (SJC_SAMPLE_1 + IJC_SAMPLE_2). For example, in the second row of the image, the value for the third replicate should be 0.06: 2/(2+31)=0.06. However, IncLevel1 = 0.037.
![SE MATS JC_2](https://github.com/user-attachments/assets/e2152ff3-c0b1-430d-b596-7901e50b0788)

**Comment by biochristmas** [comment]
Hi, @EricKutschera 
I have another question. IncLevel1 = SJC_SAMPLE_1 / (SJC_SAMPLE_1 + IJC_SAMPLE_2). For example, in the second row of the image, the value for the third replicate should be 0.06: 2/(2+31)=0.06. However, IncLevel1 = 0.037.
![SE MATS JC_2](https://github.com/user-attachments/assets/3448ac1f-fa6b-49d5-8777-f44d6376c980)

**Comment by EricKutschera** [comment]
IncLevel1 is `(IJC_SAMPLE_1/IncFormLen) / ((IJC_SAMPLE_1/IncFormLen) + (SJC_SAMPLE_1/SkipFormLen))` : https://github.com/Xinglab/rmats-turbo/issues/349

Your screenshot doesn't show the IncFormLen and SkipFormLen columns for that event, but based on the coordinates and `--readLength 150` they should be 252 and 149. Then IncLevel=(2/252)/((2/252)+(31/149)) = 0.0367

**Comment by biochristmas** [comment]
@EricKutschera 
Thank you for your prompt response. The issue has been resolved. Best regards and good luck!

**Comment by biochristmas** [comment]
Hi, @EricKutschera 
I have another question. Can RMATS extract AS events solely from the annotated GTF file? Recently, I used the astalavista software to identify AS events in a GTF file. As far as I know, this software compares transcripts pairwise under each gene in the annotation file. However, the number of AS events reported in the result file does not match the number of transcripts visualized in the GTF file using IGV. For example, the gene Bnascaffold0026T0058600ZS in IGV has 4 transcripts, and visual inspection indicates at least 3 RI events. However, the astalavista result file only identified 2 RI events.
![IGV](https://github.com/user-attachments/assets/25a09236-983b-4ed7-abd2-bb2511844102)
![astalavista](https://github.com/user-attachments/assets/c747a5a2-5ffa-4d3a-8817-c4fb1a9302ab)

**Comment by EricKutschera** [comment]
Yes rMATS can detect events just from the GTF file. See this comment which has a link to an empty bam file: https://github.com/Xinglab/rmats-turbo/issues/79#issuecomment-764704152
> You could find the set of annotated events by running rmats with a GTF file and with a mostly empty input

Based on the screenshot I think rMATS will only detect 2 RI events. See this post for details about the intron detection code: https://github.com/Xinglab/rmats-turbo/issues/17
